### PR TITLE
Track server performances in DB

### PR DESCRIPTION
**Pull request by LucasBrazi06**
Goal is to provide next a UI Performance overview of our systems with different charts to pin point issues while scaling with the number of charging stations

### CODE REVIEW

**Review by jerome-benoit — COMMENTED** [review]
I think it miss a task to cleanup old performance records and a way to disable all performance measurements to avoid the overhead.  

I would also add the serialisation of `process.memoryUsage()` to each record.

File: src/types/Performance.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { ServerAction } from './Server';
+
+export default interface Performance {
```
[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply src/types/Performance.ts]: Renamed Performance -> PerformanceRecord

File: src/utils/Utils.ts [hunk 2]
```diff
@@ -1364,6 +1365,32 @@ export default class Utils {
     };
   }
 
+  public static buildPerformance(params: {
```
[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply src/utils/Utils.ts]: Renamed Performance -> PerformanceRecord

File: src/utils/Utils.ts [hunk 3]
```diff
@@ -1364,6 +1365,32 @@ export default class Utils {
...
+  }): PerformanceRecord {
+    return {
+      tenantID: params.tenantID,
+      timestamp: new Date(),
+      durationMs: params.durationMs,
+      sizeKb: params.sizeKb,
+      host: Utils.getHostname(),
+      process: cluster.isWorker ? 'worker ' + cluster.worker.id.toString() : 'master',
+      memoryTotalGb: Utils.truncTo(os.totalmem() / 1000000000, 3),
+      memoryFreeGb: Utils.truncTo(os.freemem() / 1000000000, 3),
```
Comment (jerome-benoit) [inline src/utils/Utils.ts:1381]: I missed that: ideally only the raw value are stored in performance record to allow fine grained statistic computation later on: mean, median, holtz-winter smoothing for forecasting, etc. 
If any rounding or truncation is need, it's done later on after reading the performance record.

Comment (LucasBrazi06) [reply src/utils/Utils.ts]: Okay. Just removed truncating.

**Review by jerome-benoit — APPROVED** [review]
After the merge of the stringify I've forgotten, that PR is good to go. 

Edit: not needed, all `JSON.stringify()` I've added are unneeded.

**Pull request merged**